packages/app-builder-lib/src/targets/snap/linuxTools.ts

```diff
@@ -0,0 +1,37 @@
+import { getBinFromUrl } from "../../binDownload"
+import * as os from "os"
+import * as path from "path"
+
+async function getAppImageToolDir(): Promise<string> {
+  return await getBinFromUrl("appimage", "12.0.1", "3el6RUh6XoYJCI/ZOApyb0LLU/gSxDntVZ46R6+JNEANzfSo7/TfrzCRp5KlDo35c24r3ZOP7nnw4RqHwkMRLw==")
+}
+
+function getAppImageToolBin(toolDir: string): string {
+  return process.platform === "darwin" ? path.join(toolDir, "darwin") : path.join(toolDir, `linux-${goArchToArchSuffix()}`)
+}
+
+function goArchToArchSuffix(): string {
+  const arch = os.arch()
+  switch (arch) {
+    case "arm":
+      return "arm32"
+    default:
+      return arch
+  }
+}
+
+async function getLinuxTool(name: string): Promise<string> {
+  const toolDir = await getAppImageToolDir()
+  return path.join(getAppImageToolBin(toolDir), name)
+}
+
+export async function getMksquashfs(): Promise<string> {
+  if (!process.env.USE_SYSTEM_MKSQUASHFS) {
+    const result = process.env.MKSQUASHFS_PATH
+    if (!result) {
+      return await getLinuxTool("mksquashfs")
+    }
+    return result
+  }
+  return "mksquashfs"
```

Comment:
Wouldn't this be clearer? However, it's just a coding habit, and there's no issue if it's not changed.


```javascript
if (process.env.USE_SYSTEM_MKSQUASHFS) {
  return "mksquashfs"
}
if (process.env.MKSQUASHFS_PATH) {
  return process.env.MKSQUASHFS_PATH
}
return await getLinuxTool("mksquashfs")
```

Reply:
Yep, looks cleaner! I'll push a change